util/kube/kube.go

```diff
@@ -107,6 +107,56 @@ func MustToUnstructured(obj interface{}) *unstructured.Unstructured {
 	return uObj
 }
 
+// UnsetLabel removes our app labels from an unstructured object
+func UnsetLabel(target *unstructured.Unstructured, key string) error {
+	labels := target.GetLabels()
+	if labels == nil {
+		labels = make(map[string]string)
```

Comment:
If labels is already nil then we should just return instead of initializing an empty map and deleting nothing from it.
```
if labels == nil {
    return nil
}
```